### PR TITLE
Adding an app install email template for email-based accounts that want to use ITP.

### PR DESCRIPTION
**Pull request by alxdarksage**
See https://sagebionetworks.jira.com/browse/BRIDGE-2131

This is for the Resilience app, so it should probably be reviewed next.

### CODE REVIEW

File: test/org/sagebionetworks/bridge/services/StudyServiceMockTest.java [hunk 1]
```diff
@@ -656,6 +666,16 @@ public void loadingStudyWithoutAppInstallLinkSmsTemplateAddsADefault() {
         assertNotNull(retStudy.getAppInstallLinkSmsTemplate());
     }
     
+    @Test
+    public void loadingStudyWithoutAppInstallLinkTemplateAddsADefault() {
```
Comment (DwayneJengSage) [inline test/org/sagebionetworks/bridge/services/StudyServiceMockTest.java]: Duplicate test.

File: test/org/sagebionetworks/bridge/models/itp/IntentToParticipateTest.java [hunk 2]
```diff
@@ -26,16 +26,17 @@ public void canSerialize() throws Exception {
                 .withSignedOn(TIMESTAMP).withConsentCreatedOn(TIMESTAMP).build();
         
         IntentToParticipate itp = new IntentToParticipate.Builder().withStudyId("studyId").withPhone(PHONE)
-                .withSubpopGuid("subpopGuid").withScope(SharingScope.ALL_QUALIFIED_RESEARCHERS).withOsName("iOS")
-                .withConsentSignature(consentSignature).build();
+                .withEmail(EMAIL).withSubpopGuid("subpopGuid").withScope(SharingScope.ALL_QUALIFIED_RESEARCHERS)
+                .withOsName("iOS").withConsentSignature(consentSignature).build();
```
Comment (DwayneJengSage) [inline test/org/sagebionetworks/bridge/models/itp/IntentToParticipateTest.java:30]: Should we also check that deserializing puts the Email back in the Java Object?

File: test/org/sagebionetworks/bridge/services/IntentServiceTest.java [hunk 3]
```diff
@@ -125,40 +142,93 @@ public void submitIntentToParticipate() {
...
+        SmsMessageProvider provider = smsMessageProviderCaptor.getValue();
+        assertEquals(mockStudy, provider.getStudy());
+        assertEquals(intent.getPhone(), provider.getPhone());
+        assertEquals("this-is-a-link", provider.getSmsRequest().getMessage());
+        assertEquals("Transactional", provider.getSmsType());        
     }
     
     @Test
-    public void submitIntentToParticipateWithoutInstallLinks() {
-        IntentToParticipate intent = TestUtils.getIntentToParticipate(TIMESTAMP);
```
Comment (DwayneJengSage) [inline test/org/sagebionetworks/bridge/services/IntentServiceTest.java:138]: Did we accidentally delete this test?

**Review by DwayneJengSage — CHANGES_REQUESTED** [review]

File: test/org/sagebionetworks/bridge/services/StudyServiceMockTest.java (hunk 1 above)
Comment (alxdarksage) [reply test/org/sagebionetworks/bridge/services/StudyServiceMockTest.java]: Removed

File: test/org/sagebionetworks/bridge/models/itp/IntentToParticipateTest.java (hunk 2 above)
Comment (alxdarksage) [reply test/org/sagebionetworks/bridge/models/itp/IntentToParticipateTest.java:30]: Added

File: test/org/sagebionetworks/bridge/services/IntentServiceTest.java (hunk 3 above)
Comment (alxdarksage) [reply test/org/sagebionetworks/bridge/services/IntentServiceTest.java:138]: It seems like it. Maybe I thought installLinkCorrectlySelected() covered this, but it doesn't. Added back.

**Review by DwayneJengSage — APPROVED** [review]

**Pull request merged**